### PR TITLE
feat(public_endpoints): add public endpoints

### PR DESCRIPTION
**Pull request by KushnerykPavel**

### CODE REVIEW

File: src/main/java/trade/wayruha/whitebit/dto/Asset.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+
+@Data
+public class Asset {
+    private String name;
+    private BigDecimal unified_cryptoasset_id;
+    private boolean can_withdraw;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Asset.java]: please rename variables to follow camelCase pattern. e.g., canWithdraw.
In order to deserialize it correctly, we would need to annotate each such field with its deserialization name using @JsonAlias("can_withdraw")

File: src/main/java/trade/wayruha/whitebit/dto/Asset.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private Confirmations confirmations;
+    private Limits limits;
+    private BigDecimal currency_precision;
+    private boolean is_memo;
+
+    @Data
+    public static class Networks {
+        private List<String> deposits;
+        private List<String> withdraws;
+        private String defaultNetwork;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Asset.java]: serialization name is 'default'. Use @JsonAlias("default")

File: src/main/java/trade/wayruha/whitebit/dto/Asset.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    private boolean can_withdraw;
+    private boolean can_deposit;
+    private String min_withdraw;
+    private String max_withdraw;
+    private String maker_fee;
+    private String taker_fee;
+    private String min_deposit;
+    private String max_deposit;
+    private Networks networks;
+    private Confirmations confirmations;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Asset.java]: should be 'private Map<String, Integer> confirmations' to be parsed correctly

File: src/main/java/trade/wayruha/whitebit/dto/Fee.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class Fee {
+    @JsonAlias("is_depositable")
+    private boolean isDepositable;
+    @JsonAlias("is_withdrawal")
+    private boolean isWithdrawal;
+    private String ticker;
+    private String name;
+    private String[] providers;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Fee.java]: it's better (simpler) to use List<String> for any arrays

File: src/main/java/trade/wayruha/whitebit/dto/Fee.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static class Deposit {
+        private String max_amount;
+        private String min_amount;
+        private String fixed;
+        private Flex flex;
+    }
+
+    @Data
+    public static class Flex {
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Fee.java]: FlexibleFee would be more informative

File: src/main/java/trade/wayruha/whitebit/dto/Fee.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    private boolean isWithdrawal;
+    private String ticker;
+    private String name;
+    private String[] providers;
+    private Withdraw withdraw;
+    private Deposit deposit;
+
+    @Data
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static class Withdraw {
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Fee.java]: Withdraw and Deposit classes are equal. We should have one universal class

File: src/main/java/trade/wayruha/whitebit/dto/Fee.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        private String max_amount;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Fee.java]: we should store floating point numbers as BigDecimal (they CAN be deserialized from String)

File: src/main/java/trade/wayruha/whitebit/dto/Trade.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+import lombok.Data;
+
+import java.math.BigDecimal;
+
+@Data
+public class Trade {
+    private BigDecimal tradeID;
+    private String price;
+    private String quote_volume;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Trade.java]: it's a number, so please use BigDecimal here and in other palces

Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Trade.java]: variable name format should be camelCase

File: src/main/java/trade/wayruha/whitebit/dto/Trade.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package trade.wayruha.whitebit.dto;
+
+import lombok.Data;
+
+import java.math.BigDecimal;
+
+@Data
+public class Trade {
+    private BigDecimal tradeID;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Trade.java]: we should not have non-floating point numbers (e.g., int, long) as BigDecimal.  
it's easier to have them as int (or long for any IDs)

File: src/main/java/trade/wayruha/whitebit/dto/Trade.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    private String price;
+    private String quote_volume;
+    private String base_volume;
+    private BigDecimal trade_timestamp;
+    private String type;
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/whitebit/dto/Trade.java]: use existing enum for type

File: src/main/java/trade/wayruha/whitebit/dto/Asset.java (hunk 1 above)
Comment (KushnerykPavel) [reply src/main/java/trade/wayruha/whitebit/dto/Asset.java]: fixed

**Review by Wayruha — APPROVED** [review]

**Pull request merged**